wordpress-mobile/WordPress-Android
# Feature/update looks of comment rows

This PR updates the look of Comment List items:
- Removes status label.
- Removes date from individual comments
- Adds a Date sub-header.
- Adds a yellow dot to indicate pending comments.
- Added a divider with offset between comments.

[![Image from Gyazo](https://i.gyazo.com/077f57e1b04a336468e4c6592fd09dfc.png)](https://gyazo.com/077f57e1b04a336468e4c6592fd09dfc)

[![Image from Gyazo](https://i.gyazo.com/0edff239c348f0bb47cfb2c9a2c371a4.png)](https://gyazo.com/0edff239c348f0bb47cfb2c9a2c371a4)

[![Image from Gyazo](https://i.gyazo.com/200f19e6fb47618ef029fc3c9f5bf25b.png)](https://gyazo.com/200f19e6fb47618ef029fc3c9f5bf25b)

Behind the scenes I expanded view holders and moved a lot of logic into them, added some abstraction, implemented `DiffUtils` when loading comments.

Original plan was to move all the comment loading and management logic from adapter, but the amount of changes would be very large, so I decided to leave that for another time.

To test:
- Confirm that pending dot appears for pending comments.
- Confirm that moderation of individual comments works (and it's correctly
- Confirm that batch moderation works.

## Regression Notes
1. Potential unintended areas of impact
Comment moderation.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
With current implementation automated tests are pretty difficult.

PR submission checklist:
